src/qibolab/dummy/platform.py

```diff
@@ -1,83 +1,36 @@
 import pathlib
 
-from qibolab.components import (
-    AcquireChannel,
-    AcquisitionConfig,
-    DcChannel,
-    DcConfig,
-    IqChannel,
-    IqConfig,
-    OscillatorConfig,
-)
+from qibolab.components import AcquireChannel, DcChannel, IqChannel
 from qibolab.instruments.dummy import DummyInstrument, DummyLocalOscillator
-from qibolab.kernels import Kernels
 from qibolab.platform import Platform
-from qibolab.serialize import (
-    load_instrument_settings,
-    load_qubits,
-    load_runcard,
-    load_settings,
-)
 
 FOLDER = pathlib.Path(__file__).parent
 
 
 def create_dummy():
     """Create a dummy platform using the dummy instrument."""
+    # register the instruments
     instrument = DummyInstrument("dummy", "0.0.0.0")
+    twpa_pump = DummyLocalOscillator("twpa_pump", "0.0.0.0")
 
-    twpa_pump_name = "twpa_pump"
-    twpa_pump = DummyLocalOscillator(twpa_pump_name, "0.0.0.0")
+    platform = Platform.load(path=FOLDER, instruments=[instrument, twpa_pump])
 
-    runcard = load_runcard(FOLDER)
-    kernels = Kernels.load(FOLDER)
-
-    qubits, couplers, pairs = load_qubits(runcard)
-    settings = load_settings(runcard)
-
-    configs = {}
-    component_params = runcard["components"]
-    configs[twpa_pump_name] = OscillatorConfig(**component_params[twpa_pump_name])
-    for q, qubit in qubits.items():
+    # attach the channels
+    for q, qubit in platform.qubits.items():
```

Comment:
Following the previous comment, the second work-around also has the disadvantage that we need to hardcode qubit names in platform.py in order to create the logical channels, as we won't be able to do this loop without the `platform`.

Reply:
Indeed, but it should be much more convenient like this:
https://github.com/qiboteam/qibolab/blob/9ac2d23d35e8a36ac2b329fca51c22da5b5c03b9/src/qibolab/dummy/platform.py#L11-L36

Reply:
The only disadvantage is that the name of the channels have to be coordinated between `platform.py` and `parameters.json` "manually".

However, I believe this seems annoying only because there is "too much freedom" in the channel names. I'm already working on it in #970, and I will push in a moment.

Reply:
Thanks. Indeed this looks much better and it also solves the issue of attaching attributes for the `Qubit` as well. I will give a try on hardware starting from #983 instead.

Reply:
Now the platform in https://github.com/qiboteam/qibolab_platforms_qrc/pull/153 is updated to work with #983. Tested with the single shot routine.